website/docs/tutorials/04-configuration-file.md

```diff
@@ -35,65 +38,98 @@ network = "testnet"
 
 
 ### `[project]`
+The `[project]` section of the `protostar.toml` file allows you to specify global options and settings for your project.
 #### `protostar-version`
-This attribute is used to warn you about compatibility issues between Protostar and a project.
+This attribute is used to manifest what Protostar version should be used with your project.
+It should be set to the version of Protostar that you are using.
+If you try to use a different version of Protostar with your project, you will receive a warning and you may experience unexpected errors.
 
 #### Shared Configuration
-You can keep command arguments in the `[project]` to share them across commands. Read the [`[COMMAND]` section](#command) to learn how to configure command arguments in the configuration.
+You can specify options that are shared by multiple Protostar commands in the `[project]` section.
+For example, the [`lib-path`](/docs/cli-reference#--lib-path-path) option is used by the [`install`](/docs/cli-reference#install), [`update`](/docs/cli-reference#update), and [`remove`](/docs/cli-reference#remove) commands, and [`cairo-path`](/docs/cli-reference#--cairo-path-path) is used by the [`build`](/docs/cli-reference#build) and [`test`](/docs/cli-reference#test) commands.
 
 ### `[contracts]`
-This section allows you to refer to a contract by a contract name across Protostar features. You can also combine many files into one contract.
+The `[contracts]` section allows you to define contract names and the corresponding files that make up each contract.
+This is useful because it allows you to refer to contracts by name rather than having to specify the full file path each time.
+You can also combine multiple files into a single contract.
+Currently, the `[contracts]` section is required by the [`protostar build`](/docs/cli-reference#build) command.
+
+Here is an example of how to define a contract in the [contracts] section:
+
 ```toml
 [contracts]
 my_contract = ["src/feature_a.cairo", "src/feature_b.cairo"]   
 ```
 
-:::note
-Currently, [`protostar build`](/docs/cli-reference#build) requires this section.
-:::
-
 ### `[COMMAND]`
-Configure command arguments, to avoid passing them manually. CLI can be represented in the configuration file in the following way:
+The `[COMMAND]` section of the `protostar.toml` file allows you to specify options and settings for a specific Protostar command.
 
-```console title="CLI"
-protostar format src tests --ignore-broken
+To configure command arguments in the `protostar.toml` file, you can specify them in a `[COMMAND]` section using the following format:
+
+```toml
+[COMMAND]
+flag = true
+list = [1, 2, 3]
+path = "./src"
```

Comment:
This whole section is in my opinion weird and does not explain the configuration on real example first. It should be named something else than `[COMMAND]`, and show usage on a command only.

Reply:
>  ...does not explain the configuration on real example first

Usually, examples are below a generic concept. For example, notice how "for example" is used on the following website https://developers.google.com/tech-writing/one/paragraphs. Also, this document already starts with a big example for those who prefer such approach to learning.

> It should be named something else than [COMMAND]

Any suggestions?

> show usage on a command only.

We need to show somehow how to represent various argument types in the configuration file, and that's the purpose of this snippet.

Reply:
As a reader i would not be aware of "genericness" of this concept (which is obviously an important remark to be made in this section) at first. I'd name the section something like "Command-specific configuration" personally, and start with the real example, then perhaps going to the general idea of this kind of configuration

Reply:
> Any suggestions?

Command arguments configuration.

It is not obvious that `COMMAND` is a place holder. You can add section word to the other titles as well to be consistent. We do not name the following section `[project.PROFILE.NAME]` 

Please move shared section after the per-command section

Reply:
[put command configuration above the shared config](https://github.com/software-mansion/protostar/pull/1308/commits/58a502ac5f326449fd5abb1c4b13aae72082d019)